scex-util/src/main/scala/com/avsystem/scex/util/function/StringUtil.java

```diff
@@ -1,53 +1,78 @@
     String regexFindGroup(String value, String pattern);
 
+    @Documentation("Returns true if the string `value` matches the regular expression `pattern`.")
     boolean regexMatches(String value, String pattern);
 
+    @Documentation("Returns a string with first part of the `value` string matching the regular expression `pattern` replaced with the `replacement` argument.")
     String regexReplace(String value, String pattern, String replacement);
 
+    @Documentation("Returns a string with all parts of the `value` string matching the regular expression `pattern` replaced with the `replacement` argument.")
     String regexReplaceAll(String value, String pattern, String replacement);
 
+    @Documentation("Returns all parts of a dot-separated string following the part indicated by `from` value, e.g. string.slice('a.b.c.d.', 2) returns 'c.d.'")
     String slice(String item, int from);
 
+    @Documentation("Returns all parts of a dot-separated string following the part indicated by `from` value. `Dot` boolean value determines if a dot is added to the output, e.g. string.slice('a.b.c.d.', 2, false) returns 'c.d'")
     String slice(String item, int from, int to, boolean dot);
 
+    @Documentation("Returns the `source` string with all non-alphanumeric characters replaced with the `replacement` string.")
     String stripToAlphanumeric(String source, String replacement);
 
+    @Documentation("Returns the `source` string with it last characters indicated by the `end` string removed.")
     String removeEnd(String source, String end);
 
+    @Documentation("Returns the `source` string with it first characters indicated by the `end` string removed.")
     String removeStart(String source, String start);
 
+    @Documentation("Returns the `source` string with `InternetGatewayDevice.` or `Device.` parts removed.")
     String removeTRRoot(String source);
 
+    @Documentation("Returns a random alphanumeric string. The `length` argument indicates the number of characters.")
     String random(int length);
 
+    @Documentation("Prepends the `str` string with `padStr` (repeated or trimmed if needed). Number of characters in the output corresponds to `size`.")
     String leftPad(String str, int size, String padStr);
 
+    @Documentation("Appends `padStr` (repeated or trimmed if needed) to the `str` string. Number of characters in the output corresponds to `size`.")
     String rightPad(String str, int size, String padStr);
 
+    @Documentation("Returns a substring of `str`. The substring begins at the specified `from` index and extends to the character at the `to` index.")
     String subString(String str, int from, int to);
 
+    @Documentation("Returns an array of strings split from `str` with the given `separator`.")
     String[] split(String str, String separator);
 
+    @Documentation("Returns the `str` string with whitespaces removed.")
```

Comment:
consider adding the information that the whitespaces are removed from both ends of the string (the ones in the middle of the string are not)